cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -976,6 +983,33 @@ public void getServiceInstances(HttpRequest request, HttpResponder responder,
   }
 
   /**
+   * Return the availability (i.e. discoverable registration) status of a service.
+   */
+  @GET
+  @Path("/apps/{app-id}/services/{service-id}/available")
+  public void getServiceStatus(HttpRequest request, HttpResponder responder,
+                               @PathParam("namespace-id") String namespaceId,
+                               @PathParam("app-id") String appId,
+                               @PathParam("service-id") String serviceId) throws NotFoundException {
+    ProgramId programId = Ids.namespace(namespaceId).app(appId).service(serviceId);
+    ProgramStatus status = lifecycleService.getProgramStatus(programId);
+    if (status == ProgramStatus.STOPPED) {
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, "Service is stopped. Please start it.");
+    } else {
+      // Construct discoverable name and return 200 OK if discoverable is present. If not return 503.
+      String serviceName = ServiceDiscoverable.getName(programId);
+      EndpointStrategy endpointStrategy = new RandomEndpointStrategy(discoveryServiceClient.discover(serviceName));
+      if (endpointStrategy.pick(300L, TimeUnit.MILLISECONDS) == null) {
+        LOG.trace("Discoverable endpoint {} not found", serviceName);
+        responder.sendString(HttpResponseStatus.SERVICE_UNAVAILABLE,
+                             "Service is running but not accepting requests yet.");
```

Comment:
"yet" may be inaccurate, it may also be shutting down, in which case the correct phrasing would be "not any more". Perhaps just say "at this time".